scripts/globals/bluemagic.lua

```diff
@@ -210,45 +211,77 @@ xi.spells.blue.usePhysicalSpell = function(caster, target, spell, params)
         params.offcratiomod = caster:getStat(xi.mod.ATT)
     end
 
-    local cratio = calculatecRatio(params.offcratiomod / target:getStat(xi.mod.DEF), caster:getMainLvl(), target:getMainLvl())
-    local hitrate = calculateHitrate(caster, target)
+    params.offcratiomod = params.offcratiomod * (caster:getMerit(xi.merit.PHYSICAL_POTENCY) + 100) / 100
+    params.bonusacc     = params.bonusacc == nil and 0 or params.bonusacc
+
+    -- params.critchance will only be non-nil if base critchance is passed from spell lua
+    local nativecrit  = xi.combat.physical.calculateSwingCriticalRate(caster, target, false, 0, 0, 0)
+    params.critchance = params.critchance == nil and 0 or utils.clamp(params.critchance / 100 + nativecrit, 0.05, 0.95)
+
+    local cratio  = calculatecRatio(params.offcratiomod / target:getStat(xi.mod.DEF), caster:getMainLvl(), target:getMainLvl())
+    local hitrate = calculateHitrate(caster, target, params.bonusacc)
 
     -------------------------
     -- Perform the attacks --
     -------------------------
 
-    local hitsdone = 0
+    local hitsdone   = 0
     local hitslanded = 0
-    local finaldmg = 0
+    local finaldmg   = 0
+    local sneakIsApplicable = caster:hasStatusEffect(xi.effect.SNEAK_ATTACK) and
+                                spell:isAoE() == 0 and
+                                params.attackType ~= xi.attackType.RANGED and
+                                (caster:isBehind(target) or caster:hasStatusEffect(xi.effect.HIDE))
+    local trickAttackTarget = (caster:hasStatusEffect(xi.effect.TRICK_ATTACK) and
+                                spell:isAoE() == 0 and
+                                params.attackType ~= xi.attackType.RANGED) and
+                                caster:getTrickAttackChar(target) or nil
 
     while hitsdone < params.numhits do
         local chance = math.random()
-        if chance <= hitrate then -- it hit
+
+        if
+            sneakIsApplicable or
+            chance <= hitrate
+        then
             -- TODO: Check for shadow absorbs. Right now the whole spell will be absorbed by one shadow before it even gets here.
 
             -- Generate a random pDIF between min and max
             local pdif = math.random(cratio[1] * 1000, cratio[2] * 1000)
-            pdif = pdif / 1000
+            pdif       = pdif / 1000
+
+            if
+                sneakIsApplicable or
+                math.random() < params.critchance
+            then
+                pdif = pdif + 0.7
```

Comment:
Crits add 1.0 to pDIF 👀

Reply:
idk i copied whatever mow had in there

Reply:
I was more concerned with the critical hit rate thing